src/Compilers/CSharp/Portable/Binder/Semantics/AccessCheck.cs

```diff
@@ -327,8 +327,9 @@ private static bool IsMemberAccessible(
                 return true;
             }
 
+            // For the purpose of accessibility checks, extension members are considered to be declared within the enclosing static type
             return IsNonPublicMemberAccessible(
-                containingType,
+                containingType.IsExtension && containingType.ContainingType is { } extensionEnclosingType ? extensionEnclosingType : containingType,
```

Comment:
>containingType.IsExtension && containingType.ContainingType is { } extensionEnclosingType [](http://example.com/codeflow?start=16&length=89)

nit: consider using the same pattern you used elsewhere in this PR (`containingType is { IsExtension: true, ContainingType: { } extensionEnclosingType } ? extensionEnclosingType : containingType`) or even introducing a helper method.